log/config.go

```diff
@@ -194,65 +194,74 @@ func formatDate(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
 	enc.AppendString(string(buf))
 }
 
-func updateScopes(options *Options, core zapcore.Core, errSink zapcore.WriteSyncer) error {
-	// init the global I/O funcs
-	writeFn.Store(func(ent zapcore.Entry, fields []zapcore.Field) error {
-		err := core.Write(ent, fields)
-		if ent.Level == zapcore.FatalLevel {
-			if options.testonlyExit == nil {
-				os.Exit(1)
-			}
-			options.testonlyExit()
-		}
-		return err
-	})
-	syncFn.Store(core.Sync)
-	errorSink.Store(errSink)
-
+func updateScopes(options *Options) error {
 	// snapshot what's there
 	allScopes := Scopes()
 
-	// update the output levels of all scopes
-	levels := strings.Split(options.outputLevels, ",")
-	for _, sl := range levels {
-		s, l, err := convertScopedLevel(sl)
-		if err != nil {
-			return err
+	// update the output levels of all listed scopes
+	if err := processLevels(allScopes, options.outputLevels, func(s *Scope, l Level) { s.SetOutputLevel(l) }); err != nil {
+		return err
+	}
+
+	// update the stack tracing levels of all listed scopes
+	if err := processLevels(allScopes, options.stackTraceLevels, func(s *Scope, l Level) { s.SetStackTraceLevel(l) }); err != nil {
+		return err
+	}
+
+	// update the caller location setting of all listed scopes
+	sc := strings.Split(options.logCallers, ",")
+	overridePresent := false
+	for _, s := range sc {
+		if s == "" {
+			continue
+		} else if s == OverrideScopeName {
```

Comment:
Nit: you don't need else continuation. Might be simpler to have to two if blocks back to back.

Reply:
Done